### DISCUSSION TITLE
[Response received] Explorer (as shell) does not start on boot - 22000.527

### DISCUSSION THREAD

**Discussion by ArthurWolfhound** [valinet/ExplorerPatcher#879] [General]
I've just turned on my laptop after receiving an update, I think it was a cumulative patch. And I was greeted with fully black screen.

It took me a moment to understand that it is just explorer failing to start. And then I see a notification from EP about symbols for build being downloaded and me needing to restart explorer. I mean, users of EP mostly would know about Ctrl+Shift+Esc (though, I haven't tried Ctrl+Alt+Del, to be fair), but some help would be nice.

Is there a way to implement some kind of warning system, that if explorer does not start, there's ExplorerPatcher popping up and providing help with options: Try to restart explorer, Check for updates, Temporarily disable EP and restart original explorer.

**Comment by valinet** [comment]
Notifications don’t work when Explorer is not running, so maybe something slightly different happened…?

**Reply by ArthurWolfhound** [reply]
At that point I've already started the Task Manager, is it possible that notifications somehow partially launched with TM? Because I know for sure that explorer was dead - I've got one of typical notifications from my AV (Bitdefender) that does not use Windows Notifications. In totally wrong place.

I am sure that explorer was dead, I was sitting for like 5 minutes before the notification from AV got me to realise that it is not a black screen but an absence of the shell.

Even despite that, I can accept that I might have misunderstood something, who knows?...

**Reply by valinet** [reply]
Yeah, the thing is, Explorer is automatically restarted when it crashes, a number of times until winlogon.exe gives up.

Anyway, so, did you solve it? What was the final result, you uninstalled EP or…? I tested it yesterday on this build and it was working fine.

If every time someone encounters a problem, the only resolution is “uninstall EP” or, at most, devising some workaround to warn that the thing is crashing, these supposed bugs will never be solved. Every day someone comes with some idea for some workaround for some stuff, but no one wants to do the investigative work to determine and fix the problem, if really any. GitHub is a site for developers, that’s the expectation, nowadays on this forums 99.9999% of the posts are for support… I don’t consider things fixed if there is some pop-up that slows it down, but I can’t work on them either without any means to reproduce, crash dumps, image of the problematic OS install etc.

**Reply by ArthurWolfhound** [reply]
No, I just started explorer manually via Task Manager and it just, went on running. Will restart my laptop now to see if it fails again.
Question - do you host "symbols" elsewhere? Because it is strange that it said something akin to loaded symbols for the build 22000.527, but that does not match even your "highest" pre-release version?

> but I can’t work on them either without any means to reproduce, crash dumps, image of the problematic OS install etc.
I can understand that

**Reply by ArthurWolfhound** [reply]
Restarted the laptop. Nothing. Just works as intended.

Sorry for all of that. You're right - should've checked event viewer before writing here. Apparently dxgi.dll faulted and explorer did die. 
Weirdly enough - There's only one error message in the event viewer and I seem to be unable to get any of the files referenced in the event viewer except one, others are already missing.

[explorercrash.zip](https://github.com/valinet/ExplorerPatcher/files/8086747/explorercrash.zip) have a look if you'll have a spare minute for that, that's all that I managed to salvage from event viewer.

**Reply by valinet** [reply]
So why is the title labelled "Not EPs fault" if `Faulting module path: C:\Windows\dxgi.dll` which is exactly ExplorerPatcher...? It was indeed an EP problem.

Thanks for the Event Viewer output. It lead me to an area in the code which could produce a fault under certain circumstances, indeed. I have now fixed that. I will push an update soon. Again, thanks very much for that, hints like these help a lot in identifying and diagnosing problems. The nature of this application is prone to potentially having small mistakes like that in the code, that's why extensive testing is always welcome.

**Reply by ArthurWolfhound** [reply]
Haha, I seem to be all over the place when communicating with people. Glad to help!

**Reply by valinet** [reply]
I have written a tutorial on the matter, to help users more easily diagnose these types of faults: https://github.com/valinet/ExplorerPatcher/discussions/881
